agent/engine/dockerclient/dockerclientfactory.go

```diff
@@ -110,7 +116,15 @@ func (f *factory) getClient(version DockerVersion) (dockeriface.Client, error) {
 // findDockerVersions loops over all known API versions and finds which ones
 // are supported by the docker daemon on the host
 func findDockerVersions(endpoint string) map[DockerVersion]dockeriface.Client {
-	clients := make(map[DockerVersion]dockeriface.Client)
+	// if the client version returns a min version and api version, then use it to return
+	// Docker versions
```

Comment:
Let's make this more explicit: `use it to return versions between MinAPIVersion and APIVerison`